virtool/samples/data.py

```diff
@@ -158,29 +158,32 @@ async def find(
         )
 
     async def get(self, sample_id: str) -> Sample:
-        document = await self._db.samples.find_one({"_id": sample_id})
-
-        if not document:
-            raise ResourceNotFoundError
+        async for document in self._db.samples.aggregate(
+            [
+                {"$match": {"_id": sample_id}},
+                *lookup_nested_user_by_id(local_field="user.id")
+            ]
+        ):
+            if not document:
+                raise ResourceNotFoundError
 
-        document["caches"] = [
-            base_processor(cache)
-            async for cache in self._db.caches.find({"sample.id": sample_id})
-        ]
+            document["caches"] = [
+                base_processor(cache)
+                async for cache in self._db.caches.find({"sample.id": sample_id})
+            ]
 
-        document = await apply_transforms(
-            base_processor(document),
-            [
-                ArtifactsAndReadsTransform(self._pg),
-                AttachLabelsTransform(self._pg),
-                AttachSubtractionTransform(self._db),
-                AttachUserTransform(self._db),
-            ],
-        )
+            document = await apply_transforms(
+                base_processor(document),
+                [
+                    ArtifactsAndReadsTransform(self._pg),
+                    AttachLabelsTransform(self._pg),
+                    AttachSubtractionTransform(self._db),
```

Comment:
`AttachSubtractionTransform` should also be replaceable with an aggregation pipeline step as it is fetching data from mongo